From b406b0895a95949db2adfedaeda451f36f2b51c3 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Tue, 25 Aug 2020 10:49:11 -0400
Subject: [PATCH] systemd: switch to KillMode=mixed

KillMode=none is deprecated, so we need to stop using it.

For now, use `KillMode=mixed` and `IgnoreOnIsolate=true` instead.

In the future, we should change plymouth to be able to exit and
start again without restarting the active animation, but that's
going to require some effort.

https://gitlab.freedesktop.org/plymouth/plymouth/-/issues/123
---
 systemd-units/plymouth-start.service.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/systemd-units/plymouth-start.service.in b/systemd-units/plymouth-start.service.in
index 3d00cc6..830a62d 100644
--- a/systemd-units/plymouth-start.service.in
+++ b/systemd-units/plymouth-start.service.in
@@ -6,11 +6,12 @@ After=systemd-vconsole-setup.service systemd-udev-trigger.service systemd-udevd.
 Before=systemd-ask-password-plymouth.service
 ConditionKernelCommandLine=!plymouth.enable=0
 ConditionVirtualization=!container
+IgnoreOnIsolate=true
 
 [Service]
 ExecStart=@PLYMOUTH_DAEMON_DIR@/plymouthd --mode=boot --pid-file=@plymouthruntimedir@/pid --attach-to-session
 ExecStartPost=-@PLYMOUTH_CLIENT_DIR@/plymouth show-splash
 Type=forking
 RemainAfterExit=yes
-KillMode=none
+KillMode=mixed
 SendSIGKILL=no
-- 
2.29.2

